src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -11,55 +11,87 @@
 
 namespace Symfony\Component\Form\Extension\DependencyInjection;
 
+use Psr\Container\ContainerInterface;
 use Symfony\Component\Form\FormExtensionInterface;
 use Symfony\Component\Form\FormTypeGuesserChain;
 use Symfony\Component\Form\Exception\InvalidArgumentException;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 
 class DependencyInjectionExtension implements FormExtensionInterface
 {
-    private $container;
     private $typeServiceIds;
-    private $typeExtensionServiceIds;
     private $guesserServiceIds;
     private $guesser;
     private $guesserLoaded = false;
-
-    public function __construct(ContainerInterface $container, array $typeServiceIds, array $typeExtensionServiceIds, array $guesserServiceIds)
+    private $typeContainer;
+    private $typeExtensionServices;
+    private $guesserServices;
+
+    /**
+     * Constructor.
+     *
+     * @param ContainerInterface $typeContainer
+     * @param iterable[]         $typeExtensionServices
+     * @param iterable[]         $guesserServices
```

Comment:
`iterable[]` make me understand that the value should be an array of iterables.
`iterable|FormTypeGuesser[]` instead?